ui-tests/tst/software/aws/toolkits/jetbrains/uitests/s3/S3BrowserTest.kt

```diff
@@ -0,0 +1,200 @@
+import java.util.UUID
+
+@TestInstance(Lifecycle.PER_CLASS)
+class S3BrowserTest {
+    private val testDataPath: Path = Paths.get(System.getProperty("testDataPath"))
+
+    private val profile = "default"
+    private val credential = "Profile:$profile"
+    private val region = "Oregon (us-west-2)"
+    private val date = LocalDate.now().format(DateTimeFormatter.BASIC_ISO_DATE)
+    private val bucket = "uitest-$date-${UUID.randomUUID()}"
+    private val folder = UUID.randomUUID().toString()
+
+    private val S3 = "S3"
+    private val createBucketText = "Create S3 Bucket"
+    private val deleteBucketText = "Delete S3 Bucket"
+    private val upload = "Upload..."
+    private val newFolder = "New folder..."
+    private val rename = "Rename..."
+    private val delete = "Delete..."
+
+    private val jsonFile = "hello.json"
+    private val jsonFile2 = "hello2.json"
+    private val newJsonName = "helloooooooooo.json"
+
+    @TempDir
+    lateinit var tempDir: Path
+
+    @Test
+    @CoreTest
+    fun testS3Browser() = uiTest {
+        welcomeFrame {
+            newProject(tempDir)
+        }
+        idea {
+            waitForBackgroundTasks()
+            setCredentials(credential, region)
+            showAwsExplorer()
+        }
+        idea {
+            step("Create bucket named $bucket") {
+                awsExplorer {
+                    openExplorerActionMenu(S3)
+                }
+                find<ComponentFixture>(byXpath("//div[@text='$createBucketText']")).click()
+                find<JTextFieldFixture>(byXpath("//div[@class='JTextField']"), Duration.ofSeconds(5)).text = bucket
+                find<ComponentFixture>(byXpath("//div[@text='Create']")).click()
+            }
+
+            // Wait for the bucket to be created
+            Thread.sleep(10000)
+
+            awsExplorer {
+                step("Open editor for bucket $bucket") {
+                    expandExplorerNode(S3)
+                    doubleClickExplorer(S3, bucket)
+                }
+            }
+
+            // Click on the tree to make sure it's there + we aren't selecting anything else
+            s3Tree { click() }
+
+            step("Upload object to top-level") {
+                findAndClick("//div[@accessiblename='$upload' and @class='ActionButton']")
+                // Wait for the file selector to load, on 193 if we don't we try to upload the folder
+                // TODO FIX_WHEN_MIN_IS_201 don't need to wait on 201+
+                Thread.sleep(2000)
+                fillSingleTextField(testDataPath.resolve("testFiles").resolve(jsonFile).toString())
+                pressOk()
+                // Wait for the item to be uploaded
+                Thread.sleep(1000)
+                s3Tree {
+                    findText(jsonFile)
+                }
+            }
+
+            step("Create folder") {
+                findAndClick("//div[@accessiblename='$newFolder' and @class='ActionButton']")
```

Comment:
for some reason I want to say that there is an `anActionButon(name)` but I can't switch branches right now to go look.

Reply:
What do you mean?